src/dune_pkg/tar.ml

```diff
@@ -11,23 +11,13 @@ let bin =
 
 let output_limit = 1_000_000
 
-let temp_dir_in_build =
-  lazy
-    (let dir = Path.relative (Path.build Path.Build.root) ".temp" in
-     Path.mkdir_p dir;
-     dir)
-;;
-
 let extract ~archive ~target =
   let* () = Fiber.return () in
   let tar = Lazy.force bin in
   let target_in_temp =
     let prefix = Path.basename target in
     let suffix = Path.basename archive in
-    match target with
```

Comment:
I think this now assumes that `target` always lives in `Path.Build.t`? We should do one of the following:

1. changing target to be `Path.Build.t`
2. assert that this is the case
3. revert back to the old code for handling non build directories.

#3 would be my preference as we need code like this in other parts of dune.

Reply:
Good point, there was indeed an issue where it would write to the wrong place. I changed it to check for the build folder and only create it in the build folder in that case. However in the other case we cannot use `Temp.create` with `rename` as `Temp.create` will (on my system) always create in `/tmp` and it will always fail unless the target is in `/tmp` as well.

So I used the "subdir of parent dir" approach here as well.